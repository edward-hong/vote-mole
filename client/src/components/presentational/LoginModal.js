import React from 'react'
import { Modal, ModalBody, Button } from 'reactstrap'
import styled from 'styled-components'

import { COLOURS, AUTH_GOOGLE_LOGIN_PATH } from '../../constants'

const GoogleLoginButton = styled(Button)`
	color: ${COLOURS.google};
	border-color: ${COLOURS.google};

	&:hover,
	&:active {
		background-color: ${COLOURS.google};
		border-color: ${COLOURS.google};
	}

	&:focus {
		box-shadow: 0 0 0 3px ${COLOURS.googleShadow};
	}
`

const LoginModal = ({ isOpen, toggle }) => (
	<Modal isOpen={isOpen} toggle={toggle}>
		<ModalBody>
			<GoogleLoginButton
				href={AUTH_GOOGLE_LOGIN_PATH}
				className="btn-google"
				outline
				block>
				Login with Google
			</GoogleLoginButton>
		</ModalBody>
	</Modal>
)

export default LoginModal
