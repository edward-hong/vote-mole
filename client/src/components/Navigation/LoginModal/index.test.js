import React from 'react'
import { render } from '@testing-library/react'

import LoginModal from '.'

describe('LoginModal', () => {
	it('returns null when `isOpen` prop is false', () => {
		const { container } = render(
			<LoginModal isOpen={false} toggle={() => {}} />,
		)

		expect(container.firstChild).toBeNull()
	})

	it('renders correctly when `isOpen` props is true', () => {
		const { container } = render(<LoginModal isOpen={true} toggle={() => {}} />)

		expect(container.firstChild).toMatchSnapshot()
	})
})
