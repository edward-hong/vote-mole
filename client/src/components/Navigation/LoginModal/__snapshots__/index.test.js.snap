// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginModal renders correctly when \`isOpen\` props is true 1`] = `null`;
